core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,180 @@
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
 	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/workflow"
 	"github.com/smartcontractkit/chainlink/v2/core/services/job"
 )
 
-func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (job.ServiceCtx, error) {
+const defaultSendChannelBufferSize = 1000
+
+type Response struct {
+	Success      bool   `json:"success"`
+	ErrorMessage string `json:"error_message,omitempty"`
+}
+
+type workflowConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	connector connector.GatewayConnector
+	lggr      logger.Logger
+	mu        sync.Mutex
+	triggers  map[string]chan capabilities.TriggerResponse
+	signerKey *ecdsa.PrivateKey
+}
+
+var _ capabilities.TriggerCapability = (*workflowConnectorHandler)(nil)
+var _ services.Service = &workflowConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (job.ServiceCtx, error) {
 	// TODO (CAPPL-22, CAPPL-24):
 	//   - decode config
 	//   - create an implementation of the capability API and add it to the Registry
 	//   - create a handler and register it with Gateway Connector
 	//   - manage trigger subscriptions
 	//   - process incoming trigger events and related metadata
-	return nil, nil
+
+	handler := &workflowConnectorHandler{
+		connector: connector,
+		signerKey: signerKey,
+		lggr:      lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	// is this the right way to register with gateway connector?  Cron trigger doesn't do this.
+	err := connector.AddHandler([]string{"add_workflow"}, handler)
+
+	return handler, err
+}
+
+// from Web API Trigger Doc
+// trigger_id          - ID of the trigger corresponding to the capability ID
+// trigger_event_id    - uniquely identifies generated event (scoped to trigger_id and sender)xx
+// timestamp           - timestamp of the event, needs to be within certain freshness to be processed
+// sub_events {        - a list of per-topic-set components of this trigger event (can be of size 1)
+//   topics            - [OPTIONAL] list of topics (strings) to be started by this event (affects all topics if empty)
+//   workflow_owners   - [OPTIONAL] list of workflow owners allowed to receive this event (affects all workflows if empty)
+//   params            - key-value pairs that will be used as trigger output in the workflow Engine (translated to values.Map)
+
+//  The sample script does
+// 	workflowSpec := flag.String("workflow_spec", "[my spec abcd]", "Workflow Spec")
+// 	payloadJSON := []byte("{\"spec\": \"" + *workflowSpec + "\"}")
+//
+// how do these reconcile?
+// How do we get the TriggerID to look up in the map of TriggerIDs to connection?
```

Comment:
This lookup won't be based on Trigger ID. Every incoming HTTP message needs to be matched to subscribed workflows based on sender and topics (see design doc).

Reply:
I'll do that in CAPPL-24 if that's ok